oozie-el-extensions/src/main/java/org/apache/oozie/extensions/OozieELExtensions.java

```diff
@@ -72,15 +77,31 @@ public static String ph3_dataIn(String dataInName, String part) {
         //optional input
         if (uristr == null) {
             Element dsEle = getDSElement(eval, dataInName);
-            Configuration conf = new Configuration();
             SyncCoordAction appInst = (SyncCoordAction) eval.getVariable(CoordELFunctions.COORD_ACTION);
+
+            Configuration conf = new Configuration();
+            conf.set(OozieClient.USER_NAME, (String)eval.getVariable(OozieClient.USER_NAME));
             try {
                 ELEvaluator instEval = CoordELEvaluator.createInstancesELEvaluator(dsEle, appInst, conf);
                 StringBuilder instances = new StringBuilder();
-                CoordCommandUtils.resolveInstanceRange(dsEle, instances , appInst, conf, instEval);
-                uristr = CoordCommandUtils.createEarlyURIs(dsEle, instances.toString(),
-                        new StringBuilder(), new StringBuilder());
-                uristr = uristr.replace(CoordELFunctions.INSTANCE_SEPARATOR, ",");
+                StringBuilder urisWithDoneFlag = new StringBuilder();
+                CoordCommandUtils.resolveInstanceRange(dsEle, instances, appInst, conf, instEval);
+                CoordCommandUtils.createEarlyURIs(dsEle, instances.toString(),
+                        new StringBuilder(), urisWithDoneFlag);
+                XLog.getLog(OozieELExtensions.class).debug("Resolved instances for " + dataInName + " : "
+                        + urisWithDoneFlag.toString());
+                // Check if availability flags are present for each instance.
```

Comment:
This may not be the case for all feeds. A feed may not have a availability Flag. In which case,  mere presence of the directory is good enough.